hooks/pre/10-reset_data.rb

```diff
@@ -0,0 +1,143 @@
+# See bottom of the script for the command that kicks off the script
+require 'English'
+
+def reset
+  stop_services
+  start_services(['postgresql']) if local_postgresql?
+  empty_db_in_postgresql('foreman') if foreman_server?
+  reset_candlepin if module_enabled?('katello')
+  reset_pulp if pulp_enabled?
+  empty_db_in_postgresql('pulpcore') if pulpcore_enabled?
+end
+
+def load_db_config(db)
+  case db
+  when 'foreman'
+    module_name  = 'foreman'
+    user_param   = 'username'
+    name_param   = 'database'
+    param_prefix = 'db_'
+  when 'candlepin'
+    module_name  = 'katello'
+    user_param   = 'user'
+    name_param   = 'db_name'
+    param_prefix = 'candlepin_db_'
+  when 'pulpcore'
+    module_name  = 'foreman_proxy_content'
+    user_param   = 'user'
+    name_param   = 'db_name'
+    param_prefix = 'pulpcore_postgresql_'
+  else
+    raise "installer module unknown for db: #{db}"
+  end
+  {
+    host: param_value(module_name, "#{param_prefix}host") || 'localhost'
+    post: param_value(module_name, "#{param_prefix}port") || 5432
+    database: param_value(module_name, "#{param_prefix}#{name_param}") || db
+    username: param_value(module_name, "#{param_prefix}#{user_param}")
+    password: param_value(module_name, "#{param_prefix}password")
+  }
+end
```

Comment:
I think at this point it's sufficiently readable and simpler than loading config separately for all 3 databases.

Thanks for the feedback on it @ekohl . At this point is it satisfactory to you?

Reply:
Fixed typos...

Reply:
Yes, though I'd keep a empty line after `end` (from `case`) to have a clear visual split.

Reply:
Done.

Reply:
Also, looking at it we may want to align the parameter names at some point. I always liked `username` because it's the same length as `password`. Let's keep this in mind for the refactor.

Btw, candlepin also suggests you'll end up with `candlepin_db_db_name`.

Reply:
Do you mean alignment, as in adjusting whitespace in this file to make it more readable?

Or alignment as in, using common parameter names across the 3 modules so that the entire case block here would not be needed? Because I definitely agree, and now feels like the time to do it while we are already removing pulp2/mongodb and unifying around postgresql.

Reply:
I guess the whitespace is a nonstarter, as the linter says we cannot have nice things...

Reply:
> Or alignment as in, using common parameter names across the 3 modules so that the entire case block here would not be needed? Because I definitely agree, and now feels like the time to do it while we are already removing pulp2/mongodb and unifying around postgresql.

This.

Reply:
@ekohl agreed. Opened: https://projects.theforeman.org/issues/30388